nmodl/ode.py

```diff
@@ -291,7 +294,18 @@ def solve_non_lin_system(eq_strings, vars, constants, function_calls):
     # interweave
     code = _interweave_eqs(vecFcode, vecJcode)
 
-    return code
+    return code, linear
+
+
+def _is_linear(eqs, state_vars, sympy_vars):
+    for expr in eqs:
+        for (x, y) in itertools.combinations_with_replacement(state_vars, 2):
```

Comment:
why with replacement, do we want to try dxdx?

Reply:
Yes, dxdx is intentional.
Otherwise it would not detect that x^2 is nonlinear.

Reply:
Thanks for clarification!